src/Nats/NatsStatus.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Squadron
+{
+    /// <summary>
+    /// Status checker for Redis
+    /// </summary>
+    /// <seealso cref="IResourceStatusProvider" />
+    public class NatsStatus : IResourceStatusProvider, IDisposable
+    {
+        private readonly HttpClient _httpClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NatsStatus"/> class.
+        /// </summary>
+        /// <param name="host">Hostname</param>
+        public NatsStatus(string host)
+        {
+            Debug.Assert(
+                Uri.TryCreate($"http://{host}/", UriKind.Absolute, out Uri uri),
+                $"Bad host string '{host}'");
+
+            _httpClient = new HttpClient
+            {
+                Timeout = TimeSpan.FromSeconds(5),
+                BaseAddress = uri
+            };
+        }
+
+        /// <inheritdoc/>
+        public async Task<Status> IsReadyAsync(CancellationToken cancellationToken)
+        {
+            try {
+                HealthzResponse response = await _httpClient.GetFromJsonAsync<HealthzResponse>(
+                    new Uri("/healthz", UriKind.Relative), cancellationToken);
+
+                Debug.Assert(response?.Status == "ok", "status == 'ok'");
```

Comment:
This assert is here (not an if/else) because the health check is based on the http code being 200, not the actual status string. The status string should always be ok; the assert documents the assumption. GetFromJsonAsync will call EnsureHttpSuccessCode() internally and throw for non-200 result.